### PR TITLE
Refactor CLI entrypoint and commands registration

### PR DESCRIPTION
**Pull request by demartinofra**
Refactoring and cleaning-up the way CLI commands are registered and executed.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/cli/commands/cluster.py [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    def register_command_args(self, parser: ArgumentParser) -> None:  # noqa: D102
+        parser.add_argument("--color", action="store_true", default=False, help="Display the cluster status in color.")
+
+    def execute(self, args: Namespace, extra_args: List[str]) -> None:  # noqa: D102
+        from pcluster.cli_commands.commands import list_clusters
+
+        list_clusters(args)
+
+
+class InstancesCommand(CliCommand):
```
Comment (enrico-usai) [inline cli/src/pcluster/cli/commands/cluster.py:220]: Should we move it in a instance.py module?

File: cli/src/pcluster/cli/entrypoint.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                "console": {"format": "%(message)s"},
+            },
+            "handlers": {
+                "default": {
+                    "level": "DEBUG",
+                    "formatter": "standard",
+                    "class": "logging.handlers.RotatingFileHandler",
+                    "filename": logfile,
+                    "maxBytes": 5 * 1024 * 1024,
+                    "backupCount": 1,
```
Comment (enrico-usai) [inline cli/src/pcluster/cli/entrypoint.py:121]: not related to your patch but what about increasing the number of backups?

Comment (demartinofra) [reply cli/src/pcluster/cli/entrypoint.py]: Increased to 3

File: cli/src/pcluster/cli/commands/cluster.py [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+    def __init__(self, subparsers):
+        super().__init__(
+            subparsers, name=self.name, help=self.help, description=self.description, config_arg=True, nowait_arg=True
+        )
+
+    def register_command_args(self, parser: ArgumentParser) -> None:  # noqa: D102
+        parser.add_argument(
+            "cluster_name",
+            help="Defines the name of the cluster. The CloudFormation stack name will be "
+            "parallelcluster-[cluster_name]",
```
[the remark this answers is not in the document]
Comment (demartinofra) [reply cli/src/pcluster/cli/commands/cluster.py]: Done

File: cli/src/pcluster/cli/commands/cluster.py (hunk 1 above)
Comment (demartinofra) [reply cli/src/pcluster/cli/commands/cluster.py:221]: Yes let me do this in a later patch where I'll try to better group all commands

**Review by enrico-usai — APPROVED** [review]

**Pull request merged**